src/crimson/osd/backfill_state.cc

```diff
@@ -496,16 +509,21 @@ bool BackfillState::ProgressTracker::tracked_objects_completed() const
   return registry.empty();
 }
 
-void BackfillState::ProgressTracker::enqueue_push(const hobject_t& obj)
+bool BackfillState::ProgressTracker::enqueue_push(const hobject_t& obj)
 {
-  ceph_assert(registry.count(obj) == 0);
-  registry[obj] = registry_item_t{ op_stage_t::enqueued_push, std::nullopt };
+  const bool first_seen = registry.count(obj) == 0;
```

Comment:
this is an anti pattern using STL. how about:
```c++
[[maybe_unused]] auto [it, first_seen] = 
  registry.try_emplace(obj, op_stage_t::enqueued_push, std::nullopt);
return first_seen;
```

Reply:
Thanks for pointing this out! Fixed and repushed.